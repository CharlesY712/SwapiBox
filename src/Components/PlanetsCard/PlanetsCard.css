.card {
  display: inline-block;
  margin: 10px;
  color: #feda4a;
}

h4 {
  margin-top: 10px;
  margin-bottom: 10px;
}

p {
  margin-top: 0;
  margin-bottom: 0;
}