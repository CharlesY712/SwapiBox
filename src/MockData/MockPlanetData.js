const MockPlanetData = {
  climate: "temperate",
  name: "Alderaan",
  population: "2000000000",
  residents:["Leia Organa", "Bail Prestor Organa", "Raymus Antilles"],
  terrain: "grasslands, mountains"
};

export default MockPlanetData;