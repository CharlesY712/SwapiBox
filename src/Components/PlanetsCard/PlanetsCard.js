import React from 'react';
import './PlanetsCard.css';
import PropTypes from 'prop-types';

const PlanetsCard = ({name, terrain, population, climate, residents}) => {
  return (
    <div className='card'>
      <h1>{name}</h1>
      <h4>Terrain: <br/>{terrain}</h4>
      <h4>Population: <br/>{population}</h4>
      <h4>Climate: <br/>{climate}</h4>
      <h4>Residents: <br/>{residents.length ? 
        residents.map((resident, index) => 
          <p key={index}>{resident}</p>) : 'None'}</h4>
    </div>
  );
};

PlanetsCard.propTypes = {
  name: PropTypes.string,
  terrain: PropTypes.string,
  population: PropTypes.string,
  climate: PropTypes.string,
  residents: PropTypes.array
};

export default PlanetsCard;