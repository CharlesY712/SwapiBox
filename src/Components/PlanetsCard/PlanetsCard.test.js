import React from 'react';
import {shallow} from 'enzyme';
import PlanetsCard from './PlanetsCard';
import MockPlanetData from '../../MockData/MockPlanetData';

describe('PlanetsCard', () => {
  const mockPlanetsData = MockPlanetData;
  let wrapper = shallow(<PlanetsCard 
    name={mockPlanetsData.name}
    terrain={mockPlanetsData.terrain}
    population={mockPlanetsData.population}
    climate={mockPlanetsData.climate}
    residents={mockPlanetsData.residents}/>);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});