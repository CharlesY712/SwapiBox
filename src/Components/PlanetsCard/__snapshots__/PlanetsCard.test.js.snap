// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetsCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetsCard
    climate="temperate"
    name="Alderaan"
    population="2000000000"
    residents={
        Array [
            "Leia Organa",
            "Bail Prestor Organa",
            "Raymus Antilles",
          ]
    }
    terrain="grasslands, mountains"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Alderaan
</h1>,
        <h4>
          Terrain: 
          <br />
          grasslands, mountains
</h4>,
        <h4>
          Population: 
          <br />
          2000000000
</h4>,
        <h4>
          Climate: 
          <br />
          temperate
</h4>,
        <h4>
          Residents: 
          <br />
          <p>
                    Leia Organa
          </p>
          <p>
                    Bail Prestor Organa
          </p>
          <p>
                    Raymus Antilles
          </p>
</h4>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Alderaan",
        },
        "ref": null,
        "rendered": "Alderaan",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Terrain: ",
            <br />,
            "grasslands, mountains",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Terrain: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "grasslands, mountains",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Population: ",
            <br />,
            "2000000000",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Population: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "2000000000",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Climate: ",
            <br />,
            "temperate",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Climate: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "temperate",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Residents: ",
            <br />,
            Array [
              <p>
                Leia Organa
</p>,
              <p>
                Bail Prestor Organa
</p>,
              <p>
                Raymus Antilles
</p>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "Residents: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "Leia Organa",
            },
            "ref": null,
            "rendered": "Leia Organa",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Bail Prestor Organa",
            },
            "ref": null,
            "rendered": "Bail Prestor Organa",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "Raymus Antilles",
            },
            "ref": null,
            "rendered": "Raymus Antilles",
            "type": "p",
          },
        ],
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Alderaan
</h1>,
          <h4>
            Terrain: 
            <br />
            grasslands, mountains
</h4>,
          <h4>
            Population: 
            <br />
            2000000000
</h4>,
          <h4>
            Climate: 
            <br />
            temperate
</h4>,
          <h4>
            Residents: 
            <br />
            <p>
                        Leia Organa
            </p>
            <p>
                        Bail Prestor Organa
            </p>
            <p>
                        Raymus Antilles
            </p>
</h4>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Alderaan",
          },
          "ref": null,
          "rendered": "Alderaan",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Terrain: ",
              <br />,
              "grasslands, mountains",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Terrain: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "grasslands, mountains",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population: ",
              <br />,
              "2000000000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Population: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "2000000000",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Climate: ",
              <br />,
              "temperate",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Climate: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "temperate",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Residents: ",
              <br />,
              Array [
                <p>
                  Leia Organa
</p>,
                <p>
                  Bail Prestor Organa
</p>,
                <p>
                  Raymus Antilles
</p>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "Residents: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "Leia Organa",
              },
              "ref": null,
              "rendered": "Leia Organa",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "Bail Prestor Organa",
              },
              "ref": null,
              "rendered": "Bail Prestor Organa",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "Raymus Antilles",
              },
              "ref": null,
              "rendered": "Raymus Antilles",
              "type": "p",
            },
          ],
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
